icu4c/source/common/unicode/brkiter.h

```diff
@@ -93,6 +95,34 @@ U_NAMESPACE_BEGIN
  * are available at http://www.unicode.org/reports/tr14/ and
  * http://www.unicode.org/reports/tr29/.
  * <p>
+ * A boundary logically occurs between two adjacent code points. The boundary
+ * analysis functions in this file return the index of the code unit directly
+ * after a boundary - so a value of 0 means "the beginning of the string" and a
+ * value of u_strlen(string) means "the end of the string".
+ * <p>
+ * A span of text between two adjacent boundaries is called a "segment."
+ * The boundary analysis functions in this file behave correctly regardless of
```

Comment:
A break iterator's position is always on a boundary; there is no mechanism to set it to an arbitrary non-boundary position.
The functions `preceding(position)` and `following(position)` do take any position as an argument, but always leave the iterator on a nearby boundary.